eth_utils/abi.py

```diff
@@ -1,15 +1,27 @@
 from typing import (
     Any,
     Dict,
+    Union,
+)
+
+from typing_extensions import (
+    deprecated,
 )
 
 from .crypto import (
     keccak,
 )
 
 
+@deprecated(
+    "`collapse_if_tuple` is deprecated, use "
+    "`get_normalized_abi_component_type` instead."
+)
 def collapse_if_tuple(abi: Dict[str, Any]) -> str:
```

Comment:
You might consider making this method an alias of `get_normalized_abi_component_type`